### PR TITLE
Refactor export

### PR DESCRIPTION
**Pull request by smeeckaert**
I did a refactoring of the export form method.

I separated the format logic with the data fetching. I created a class Helper_Export that returns headers and values of a specific form, and i used this class into the action_export of the controller.

This will allow anyone to add easily a new export format on top of noviusos_form.

### CODE REVIEW

File: classes/controller/admin/form.ctrl.php [hunk 1]
```diff
@@ -303,61 +271,29 @@ public function action_export($id)
             gc_enable();
 
             // Send HTTP headers for inform the browser that it will receive a CSV file
-            \Response::forge(\Format::forge($csv)->to_csv()."\n", 200, array(
-                'Content-Type' => 'application/csv',
-                'Content-Disposition' => 'attachment; '.
-                    'filename='.\Nos\Orm_Behaviour_Virtualname::friendly_slug($this->item->form_name).'.csv;',
+            \Response::forge(\Format::forge($csv)->to_csv() . "\n", 200, array(
+                'Content-Type'              => 'text/plain',
```
Comment (jguyomard) [inline classes/controller/admin/form.ctrl.php]: Can be removed ;-)

File: classes/helper/export.php [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+        $offset = 0;
+        $limit  = 500;
+        foreach ($answers as $answer) {
+            $values = array();
+            foreach ($answer->fields as $answer_field) {
+                $values[$answer_field->anfi_field_id] = $answer_field;
+            }
+
+            $csv_row = array();
```
Comment (jguyomard) [inline classes/helper/export.php]: Rename variable csv_\* ?

**Comment by jguyomard** [comment]
Nice work :+1:

**Comment by Meroje** [comment]
> This will allow anyone to add easily a new export format on top of noviusos_form.

csv is still hardcoded, or am I missing something ?

**Comment by smeeckaert** [comment]
Yup, the csv export is still shipped with the application. But the logic between the data fetching and formating has been separated. That way you can extend the application as you like without having to duplicate the logic.

**Comment by Meroje** [comment]
Ah, I thought the helper was meant to be extended not the controller.

:+1:

**Pull request merged**